crates/typst/src/introspection/counter.rs

```diff
@@ -361,6 +360,12 @@ impl Counter {
         /// style exists.
         #[default]
         numbering: Option<Numbering>,
+        /// The location at which to display the counter. If omitted, displays
+        /// the counter at the location where the returned content is inserted
+        /// into the document. A suitable location can be retrieved from
+        /// [`locate`]($locate) or [`query`]($query).
+        #[default]
+        location: Option<Location>,
```

Comment:
another optional positional argument might make the API too confusing. I would also prefer to discuss this argument separately from the rest of the PR.